integration_tests/gruntz_demo3.py

```diff
@@ -0,0 +1,184 @@
+from lpython import S, str
+from sympy import Symbol, Pow, sin, oo, pi, E, Mul, Add
+
+def mrv(e: S, x: S) -> list[S]:
+    """
+    Calculate the MRV set of the expression.
+
+    Examples
+    ========
+
+    >>> mrv(log(x - log(x))/log(x), x)
+    {x}
+
+    """
+
+    if e == x:
+        list1: list[S] = [x]
+        return list1
+    if e.func == Mul or e.func == Add:
+        a: S = e.args[0]
+        b: S = e.args[1]
+        ans1: list[S] = mrv(a, x)
+        ans2: list[S] = mrv(b, x)
+        return mrv_max(mrv(a, x), mrv(b, x), x)
+    if e.func == Pow:
+        base: S = e.args[0]
+        list3: list[S] = mrv(base, x)
+        return list3
+    if e.is_Function:
+        return reduce(lambda a, b: mrv_max(a, b, x), (mrv(a, x) for a in e.args))
+    raise NotImplementedError(f"Can't calculate the MRV of {e}.")
+
+def mrv_max(f: list[S], g: list[S], x: S) -> list[S]:
+    """Compute the maximum of two MRV sets.
+
+    Examples
+    ========
+
+    >>> mrv_max({log(x)}, {x**5}, x)
+    {x**5}
+
+    """
+
+    if len(f) == 0:
+        return g
+    elif len(g) == 0:
+        return f
+    else:
+        return f | g
```

Comment:
`mrv_max` as a function is not too complex. If any list is empty return the other list and if both are non empty, just take union and return. So supporting some kind of `union` (or maybe `append`) for symbolic list (`list[S]`) might be required.